jsapp/js/components/projectDownloads/anonymousExports.es6

```diff
@@ -0,0 +1,164 @@
+    this.prepareFetchInterval(exportData.uid, exportData.status);
+
+    if (exportData.status === EXPORT_STATUSES.complete) {
+      this.setState({
+        isPending: false,
+        exportUrl: exportData.result,
+      });
+      this.downloadUrl(this.state.exportUrl);
+    }
+  }
+
+  onSubmit() {
+    if (this.state.exportUrl) {
+      // we remember the current type download to not make multiple calls
+      this.downloadUrl(this.state.exportUrl);
+    } else {
+      this.setState({isPending: true});
+
+      const defaultExportFormat = getContextualDefaultExportFormat(this.props.asset);
+
+      // NOTE: this wouldn't work for legacy formats, but luckily we don't allow
+      // choosing legacy types in this component
+      actions.exports.createExport(
+        this.props.asset.uid,
+        {
+          type: this.state.selectedExportType.value,
+          fields_from_all_versions: DEFAULT_EXPORT_SETTINGS.INCLUDE_ALL_VERSIONS,
+          group_sep: DEFAULT_EXPORT_SETTINGS.GROUP_SEPARATOR,
+          hierarchy_in_labels: DEFAULT_EXPORT_SETTINGS.INCLUDE_GROUPS,
+          lang: defaultExportFormat.value,
+          multiple_select: DEFAULT_EXPORT_SETTINGS.EXPORT_MULTIPLE.value,
+        }
+      );
+    }
+  }
+
+  prepareFetchInterval(exportUid, exportStatus) {
+    // if the export is not complete yet, and there is no fetch interval
+    // fetch it in some time again and again
+    if (
+      exportStatus !== EXPORT_STATUSES.error &&
+      exportStatus !== EXPORT_STATUSES.complete &&
+      !this.fetchIntervalId
+    ) {
+      const intervalId = setInterval(this.fetchExport.bind(this, exportUid), EXPORT_REFRESH_TIME);
+      this.fetchIntervalId = intervalId;
+    }
+
+    // clean up after it is completed
+    if (
+      exportStatus === EXPORT_STATUSES.error ||
+      exportStatus === EXPORT_STATUSES.complete
+    ) {
+      this.removeFetchInterval();
+    }
+  }
+
+  removeFetchInterval() {
+    clearInterval(this.fetchIntervalId);
+    this.fetchIntervalId = null;
+  }
+
+  fetchExport(exportUid) {
+    actions.exports.getExport(this.props.asset.uid, exportUid);
+  }
+
+  downloadUrl(url) {
+    const aEl = document.createElement('a');
+    const splitUrl = url.split('/');
+    const fileName = splitUrl[splitUrl.length - 1];
+    aEl.href = url;
+    aEl.setAttribute('download', fileName);
+    aEl.click();
+  }
+
+  /**
+   * We allow only one pending download at a time, so we disable the type
+   * selector and the export button for simplicity.
+   */
+  render() {
+    const buttonModifiers = ['blue'];
+    if (this.state.isPending) {
+      buttonModifiers.push('pending');
+    }
+
+    return (
+      <bem.FormView__cell m={['box', 'padding']}>
+        <bem.ProjectDownloads__anonymousRow>
+          <bem.ProjectDownloads__exportsSelector>
+            <ExportTypeSelector
+              disabled={this.state.isPending}
+              noLegacy
+            />
+          </bem.ProjectDownloads__exportsSelector>
+
```

Comment:
There's a lot of white space between the `Select` and the `Export` button, for a possible next release it may be helpful to add a banner that explains what is happening. Maybe it's also very self-explanatory :stuck_out_tongue:

Reply:
I just wanted to be consistent with non-anonymous UI. But it will be better with button closer to selector - fixed it :)